PathOfBuildingCommunity/PathOfBuilding
# Adds support for the "%d more recovery if used while on low life" Life flask mod

### Description of the problem being solved:
Fixes an issue with the Ikiaho's promise unique where the "Life Flasks used while on Low Life apply Recovery Instantly" mod was checking for low mana instead of low life.

Adds support for the "%d more recovery if used while on low life" mod on life flasks.

### Steps taken to verify a working solution:
- Create a life flask with the "%d more recovery if used while on low life" modifier.
- Enable the life flask.
- Tick "Are you always on Low Life?" in the config or reserve over 50% of your life.
- Check if the Life Recovery changes to the appropriate value when ticking low life.

### Link to a build that showcases this PR:
https://pobb.in/TZqJXLdYHrbd

### Before screenshot:
![image](https://github.com/PathOfBuildingCommunity/PathOfBuilding/assets/28728694/90d19448-a611-483e-8489-cfeaaedac657)

### After screenshot:
![image](https://github.com/PathOfBuildingCommunity/PathOfBuilding/assets/28728694/97dceb2c-0deb-430b-8670-4dfea0e7fab0)
![image](https://github.com/PathOfBuildingCommunity/PathOfBuilding/assets/28728694/82b6a830-a49b-484e-84a9-2b7095523845)

All calculations were tested ingame to verify results. 
Resulting values in PoB are slightly off due to rounding errors (under 1 hp/sec off).
